src/object_gui.cpp

```diff
@@ -543,9 +543,39 @@ class BuildObjectWindow : public Window {
 
 	void OnPlaceObject(Point pt, TileIndex tile) override
 	{
-		ObjectClass *objclass = ObjectClass::Get(_selected_object_class);
-		Command<CMD_BUILD_OBJECT>::Post(STR_ERROR_CAN_T_BUILD_OBJECT, CcPlaySound_CONSTRUCTION_OTHER,
-				tile, objclass->GetSpec(_selected_object_index)->Index(), _selected_object_view);
+		const ObjectSpec *spec = ObjectClass::Get(_selected_object_class)->GetSpec(_selected_object_index);
+
+		if (_settings_game.construction.build_object_area_permitted && spec->size == 0x11) {
+			VpStartPlaceSizing(tile, VPM_X_AND_Y, DDSP_BUILD_OBJECT);
+		} else {
+			Command<CMD_BUILD_OBJECT>::Post(STR_ERROR_CAN_T_BUILD_OBJECT, CcPlaySound_CONSTRUCTION_OTHER,
+				tile, spec->Index(), _selected_object_view);
```

Comment:
line continuations should be double indented, but I'd say don't have the linebreak at all